apiserver/facades/client/sshclient/facade.go

```diff
@@ -291,3 +310,37 @@ func (facade *Facade) getExecSecretToken(cloudSpec environscloudspec.CloudSpec,
 	}
 	return broker.GetSecretToken(k8sprovider.ExecRBACResourceName)
 }
+
+// getVirtualHostnameForEntity returns the virtual hostname for the given entity. It parses the tag string to
+// evaluate if the entity is a machine or a unit. If the entity is a unit, it also takes an optional container
+// name which is used to construct the virtual hostname.
+func getVirtualHostnameForEntity(modelUUID string, tagString string, container *string) (string, error) {
+	tag, err := names.ParseTag(tagString)
+	if err != nil {
+		return "", errors.Trace(err)
+	}
+	var info virtualhostname.Info
+	switch tag := tag.(type) {
```

Comment:
```suggestion
	switch tag.Kind() {
```

Then 
` case names.UnitTagKind: `  is a more used pattern in Juju.